lib/plugins/aws/deploy/compile/events/apiGateway/index.js

```diff
@@ -36,7 +36,8 @@ class AwsCompileApigEvents {
         _.forEach(this.serverless.service.functions, functionObj => {
           if (functionObj.events) {
             functionObj.events.forEach(event => {
-              if (event.http) noEndpoints = false;
+              // Allow events with empty http event to validate function
+              if ({}.hasOwnProperty.call(event, 'http')) noEndpoints = false;
```

Comment:
Thanks for the clarification. I think this would be a good scenario to disable that rule in the interest of simpler, clearer code.